ideas/new-ops.txt

```diff
@@ -0,0 +1,117 @@
+# Proposal for new stdlib operations
+
+This is a proposal for several APIs that could be added to stdlib.
+
+## `builtin` candidates
+
+Entity	Signature	Notes
+
+`Not`	`Not(x bool) (x bool)`	Return logical not operation applied to input `x`
+`Pair<A, B>`	`struct { first A; second B }`	A generic pair of items. Can stand in for other pair types we've introduced such as `ZipResult`
+`OrDefault<T>`	`OrDefault<T>(t T)`	Try to return `t` or `default` if `t` is unavailable. `default` can be injected with `IInjectable<T>`
+`Unzip<T, R>`	`Unzip<T, R>(seq stream<ZipResult<T, R>>) (first stream<T>, second stream<R>)`	The inverse of `Zip<T, R>` which is already in `builtin`
+
+## DI Interfaces
+
+Inject these no-inport interfaces rather than having it in the signature.
```

Comment:
already discussed on Discord but for the history will copy here

any component must have at least 1 inport (and 1 outport) just so we can control it's execution

> without inport it's not possible to "trigger" it, it can only work like New - unconditionally in an infinite loop from the program start
> :sig -> (',' -> :out)
>
> doesn't have this problem